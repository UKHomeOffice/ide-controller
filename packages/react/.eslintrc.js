// @todo add vs code format on save wiki/ readme on github
module.exports = {
  env: {
    browser: true,
    es2020: true,
    node: true,
    jest: true,
  },
  extends: ['plugin:react/recommended', 'airbnb', 'prettier'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 11,
    sourceType: 'module',
  },
  plugins: ['react', 'prettier'],
  ignorePatterns: ['/src/opencv'],
  rules: {
    quotes: ['error', 'single'],
    'prettier/prettier': [
      'error',
      {
        trailingComma: 'es5',
        singleQuote: true,
        printWidth: 80,
      },
    ],
  },
};
