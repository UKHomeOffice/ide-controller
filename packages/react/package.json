{
  "name": "@ide-controller/react",
  "version": "1.6.0",
  "private": true,
  "homepage": "./",
  "author": "Viable Data",
  "scripts": {
    "dev": "BROWSER=none CI=true react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --watchAll=false",
    "test:watch": "react-scripts test",
    "eject": "react-scripts eject",
    "lint": "npx eslint --ext .js,.jsx .",
    "lint:fix": "eslint --ext .js,.jsx . --fix"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@tensorflow-models/posenet": "^2.2.1",
    "@tensorflow/tfjs": "^2.7.0",
    "govuk-frontend": "^3.8.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.3"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "jest-canvas-mock": "^2.3.0",
    "node-sass": "^4.14.1",
    "prettier": "^2.1.1"
  }
}
