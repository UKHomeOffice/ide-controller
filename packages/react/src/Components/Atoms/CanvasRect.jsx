// Global imports
import PropTypes from 'prop-types';
import React, { forwardRef, useEffect } from 'react';

// Local imports
import { isEmpty } from '../../helpers/common';
import { paintRec } from '../../helpers/canvas';

const CanvasRect = forwardRef(
  ({ coordinate, width, height, className }, canvasRef) => {
    const rePaintRec = () => {
      if (isEmpty(canvasRef)) return;
      const context = canvasRef.current.getContext('2d');
      context.clearRect(0, 0, width, height);
      paintRec(context, coordinate);
    };
    useEffect(rePaintRec, [coordinate]);

    return (
      <canvas
        ref={canvasRef}
        width={width}
        height={height}
        className={className}
      />
    );
  }
);

CanvasRect.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  coordinate: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
    width: PropTypes.number,
    height: PropTypes.number,
  }),
  className: PropTypes.string,
};

CanvasRect.defaultProps = {
  width: 100,
  height: 100,
  coordinate: {
    x: 0,
    y: 0,
    width: 100,
    heigh: 100,
  },
  className: null,
};

export default CanvasRect;
