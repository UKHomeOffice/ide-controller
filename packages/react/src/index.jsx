// Global imports
import React from 'react';
import ReactDOM from 'react-dom';
import { initAll } from 'govuk-frontend';
import 'govuk-frontend/govuk/all.scss';

// Local imports
import App from './App';
import './Components/Style/atoms.scss';
import './Components/Style/global.scss';
import './Components/Style/molecules.scss';
import './helpers/ipcMainEvents';

document.body.className = document.body.className
  ? `${document.body.className} js-enabled`
  : 'js-enabled';

initAll();

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
