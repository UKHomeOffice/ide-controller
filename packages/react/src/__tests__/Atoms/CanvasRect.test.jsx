// Global imports
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-canvas-mock';

// Local imports
import CanvasRect from '../../Components/Atoms/CanvasRect';

describe('<CanvasRect />', () => {
  const ref = {
    current: document.createElement('canvas'),
  };

  const coordinate = {
    x: 0,
    y: 0,
    width: 100,
    heigh: 100,
  };

  it('renders and match the snapshot', () => {
    const { asFragment } = render(
      <CanvasRect
        ref={ref}
        width={100}
        height={100}
        className="testClassName"
        coordinate={coordinate}
      />
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('has className equals to passed className prop', () => {
    const className = 'testClass';
    render(<CanvasRect ref={ref} className={className} />);
    const canvas = screen.getByTestId('atoms-canvas-rect');

    expect(canvas).toHaveClass(className);
  });

  it('width attribute equals to width prop', () => {
    const width = 400;
    render(<CanvasRect ref={ref} width={width} />);
    const canvas = screen.getByTestId('atoms-canvas-rect');

    expect(canvas).toHaveAttribute('width', String(width));
  });

  it('height attribute equals to height prop', () => {
    const height = 800;
    render(<CanvasRect ref={ref} height={height} />);
    const canvas = screen.getByTestId('atoms-canvas-rect');

    expect(canvas).toHaveAttribute('height', String(height));
  });
});
