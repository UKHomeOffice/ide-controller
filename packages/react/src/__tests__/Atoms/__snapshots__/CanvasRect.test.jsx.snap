// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CanvasRect /> renders and match the snapshot 1`] = `[Function]`;
